api/apps/api/src/modules/scenarios/locks/lock.service.ts

```diff
@@ -0,0 +1,58 @@
+import { Injectable } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+import { Connection, Repository } from 'typeorm';
+
+import { ScenarioLockEntity } from '@marxan-api/modules/scenarios/locks/scenario.lock.entity';
+import { LockedScenarioError } from './errors/locked.scenario.error';
+import { LockNotFoundError } from './errors/lock.not.found.error';
+
+@Injectable()
+export class LockService {
+  constructor(
+    @InjectRepository(ScenarioLockEntity)
+    private readonly locksRepo: Repository<ScenarioLockEntity>,
+    private connection: Connection,
```

Comment:
You can acquire EntityManager directly from repo, reaching to connection may not be necessary.

Reply:
https://github.com/Vizzuality/marxan-cloud/pull/673/commits/4089893cc9412c2ea89e98dd68326c806486fa6d